CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.8)
```

Comment:
I think this version of CMake does not support Boost 1.70.

Somewhen I had that problem, I think with a recent boost version (I might have built it from source myself) on a not so recent version of Ubuntu, and somehow I ended in this thread that mentions which CMake versions support which Boost versions (rule of thumb is always use a CMake version which was released later than the Boost libs version you are using):

https://stackoverflow.com/questions/42123509/cmake-finds-boost-but-the-imported-targets-not-available-for-boost-version

According to that, we should bump this to 3.14 to support Boost 1.70.

Reply:
I believe this is only true if you aren’t using the boost cmake integration (i.e. the way we are currently doing it), but I will confirm that this fork actually builds with cmake 3.8

Reply:
Ok, that makes sense to me.

Reply:
That took longer to finish than I had hoped.
I found some indication, though now I can't remember what it was, that we require cmake 3.10 on at least some systems. Of course, I just did a successful build with 3.8 on linux.

What is a larger concern, and I will update the PR description for this, is that it seems like we need boost>=1.73. I tracked this down using the conda-forge boost builds and need to confirm with my own boost builds.